src/containers/Article.js

```diff
@@ -246,6 +260,11 @@ class Article extends Component {
     return top
   }
 
+  _onResize() {
+    this._setArticleBounding()
+    viewportHeight = window.innerHeight
```

Comment:
Since you already listen on `resize` event, you can know the viewport width to decide to display the tools.
Hence, you can skip the calculation we don't need. For example, you can skip calculation for mobile device when the user is using desktop.